app/main.py

```diff
@@ -1 +1,35 @@
-# write your code here
+from __future__ import annotations
+
+
+class Animal:
+
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.hidden = False
+        self.health = health
+        self.__class__.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}," \
+               f" Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        if self.hidden is True:
+            self.hidden = False
+        else:
+            self.hidden = True
+
+
+class Carnivore(Animal):
+
+    def bite(self, animal: Animal) -> None:
+        if isinstance(animal, Herbivore) and animal.hidden is False:
```

Comment:
il likt this more `and not animal.hidden`

Reply:
Done